### PR TITLE
[DC-2113] Update findTemplate to fallback to English template

### PR DESCRIPTION
**Pull request by lxol**

### CODE REVIEW

File: app/uk/gov/hmrc/hmrcemailrenderer/templates/TemplateLocator.scala [hunk 1]
```diff
@@ -118,6 +118,15 @@ trait TemplateLocator {
   lazy val all: Seq[MessageTemplate] = templateGroups.values.flatten.toSeq
 
   def findTemplate(templateId: String): Option[MessageTemplate] =
-    all.find(_.templateId == templateId)
+    all.find(_.templateId == templateId) orElse  {
+      if (templateId.endsWith(TemplateLocator.WELSH_SUFFIX)) {
```
Comment (OlegEfrem) [inline app/uk/gov/hmrc/hmrcemailrenderer/templates/TemplateLocator.scala]: do you need this if :) ? 
can't you just do or else all.find(_.templateId == templateId.stripSuffix(...)) ?

File: test/uk/gov/hmrc/hmrcemailrenderer/templates/TemplateLocatorSpec.scala [hunk 2]
```diff
@@ -24,6 +24,16 @@ import uk.gov.hmrc.play.test.UnitSpec
...
+  "The template locator when requested Welsh template" should {
+
+    "return Welsh version of template if it exists" in new TestCase {
+      templateLocatorWithWelsh.findTemplate("template-templateGroup-1-2.cy").get.templateId shouldBe "template-templateGroup-1-2.cy"
+    }
+    "return English version of template if Welsh virsion doesn't exists  " in new TestCase {
+      templateLocator.findTemplate("template-templateGroup-1-2.cy").get.templateId shouldBe "template-templateGroup-1-2"
+    }
+  }
+
```
Comment (OlegEfrem) [inline test/uk/gov/hmrc/hmrcemailrenderer/templates/TemplateLocatorSpec.scala]: do we have a return None case?

File: app/uk/gov/hmrc/hmrcemailrenderer/templates/TemplateLocator.scala (hunk 1 above)
Comment (lxol) [reply app/uk/gov/hmrc/hmrcemailrenderer/templates/TemplateLocator.scala]: yes, makes sense.

File: test/uk/gov/hmrc/hmrcemailrenderer/templates/TemplateLocatorSpec.scala (hunk 2 above)
Comment (lxol) [reply test/uk/gov/hmrc/hmrcemailrenderer/templates/TemplateLocatorSpec.scala]: yes, done.

File: app/uk/gov/hmrc/hmrcemailrenderer/templates/TemplateLocator.scala [hunk 3]
```diff
@@ -118,6 +118,11 @@ trait TemplateLocator {
   lazy val all: Seq[MessageTemplate] = templateGroups.values.flatten.toSeq
 
   def findTemplate(templateId: String): Option[MessageTemplate] =
-    all.find(_.templateId == templateId)
+    all.find(_.templateId == templateId) orElse  {
```
Comment (OlegEfrem) [inline app/uk/gov/hmrc/hmrcemailrenderer/templates/TemplateLocator.scala]: now that you changed, it is looks like we could simplify it even further :)
guess we can just do in one find: all.find(id => id == templateId || id == templateId.stripSuffix(...)), what do you think :) ?

**Pull request merged**